lib/msprime.h

```diff
@@ -63,36 +63,27 @@
 #define MSP_PED_STATE_CLIMBING 1
 #define MSP_PED_STATE_CLIMB_COMPLETE 2
 
-/* FIXME: Using these typedefs to keep the diff size small on the initial
- * tskit transition. Can remove later. */
-typedef tsk_id_t population_id_t;
-typedef tsk_id_t node_id_t;
-typedef tsk_id_t mutation_id_t;
-typedef tsk_id_t site_id_t;
-typedef tsk_id_t label_id_t;
-
 typedef struct segment_t_t {
-    /* TODO change to population */
-    population_id_t population_id;
-    label_id_t label;
+    tsk_id_t population;
```

Comment:
I'm a bit hesitant about changing ``population_id_t`` and ``label_id_t``, actually, because of this struct. Once the dust settles on #1138, we might be able to make this struct a bit smaller by making population and label 16 bit (which is plenty big enough). This struct is important as we do have gazillions of them in large simulations. But, given that most things are 8 byte aligned these days I don't know if that would make any difference anyway.

Reply:
In general, makes sense to me to keep typedefs, like population_id_t and label_id_t. So I'll update to KEEP all the tsk_id_t typedefs that I initially removed in this PR.

Reply:
Getting rid of ``node_id_t``,  ``mutation_id_t`` and  ``site_id_t`` is good, they're definitely pointless.

Reply:
Good to know, I'll keep those of those three pointlessly precise typedefs out.